src/Features/Core/Portable/CodeRefactorings/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.InvertIf
+{
+    internal abstract partial class AbstractInvertIfCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        private const string LongLength = "LongLength";
+
+        private static readonly Dictionary<BinaryOperatorKind, BinaryOperatorKind> s_negatedBinaryMap =
+            new Dictionary<BinaryOperatorKind, BinaryOperatorKind>
+        {
+            {BinaryOperatorKind.Equals, BinaryOperatorKind.NotEquals},
+            {BinaryOperatorKind.NotEquals, BinaryOperatorKind.Equals},
+            {BinaryOperatorKind.LessThan, BinaryOperatorKind.GreaterThanOrEqual},
+            {BinaryOperatorKind.GreaterThan, BinaryOperatorKind.LessThanOrEqual},
+            {BinaryOperatorKind.LessThanOrEqual, BinaryOperatorKind.GreaterThan},
+            {BinaryOperatorKind.GreaterThanOrEqual, BinaryOperatorKind.LessThan},
+            {BinaryOperatorKind.Or, BinaryOperatorKind.And},
+            {BinaryOperatorKind.And, BinaryOperatorKind.Or},
+            {BinaryOperatorKind.ConditionalOr, BinaryOperatorKind.ConditionalAnd},
+            {BinaryOperatorKind.ConditionalAnd, BinaryOperatorKind.ConditionalOr},
+        };
```

Comment:
why do you need this?

Reply:
I'm using it to get the opposite operator of a given binary operator.  This could also be done in a method with a select statement.  Or, any other suggestions?

Reply:
Ignore my comment. I thought I was looking at a separate file